core/src/org/samtools/htsjdk/core/Locatable.java

```diff
@@ -0,0 +1,140 @@
+package org.samtools.htsjdk.core;
+
+import java.util.OptionalLong;
+
+/**
+ * Represents the location of a sequence with respect to another sequence (reference).
+ *
+ * <p>
+ * This interface represents a location or "mapping" with respect to a reference sequence. Examples
+ * of this abstraction are features located on the genome or alignment between two sequences (query
+ * with respect to a reference).
+ * </p>
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ * @implSpec note that 0-length intervals are allowed in an implementation-dependent manner.
+ */
+public interface Locatable {
+
+    /**
+     * Represents the reference name for unplaced locations.
+     *
+     * @see #getReferenceName()
+     */
+    public static final String UNPLACED_REFERENCE_NAME = "*";
+
+    /**
+     * Gets the reference name where this location is placed.
+     *
+     * <p>
+     * {@link Locatable} should either be placed on a reference sequence or being unplaced
+     * ({@link #UNPLACED_REFERENCE_NAME}).
+     * </p>
+     *
+     * @return name of the reference for the location; {@link #UNPLACED_REFERENCE_NAME} if unplaced.
+     * Never {@code null}
+     */
+    String getReferenceName();
+
+    /**
+     * Gets the start position on the reference (1-based).
+     *
+     * <p>
+     * {@link Locatable} should have a non-negative location on the reference sequence; if unplaced
+     * ({@code getReferenceName() == UNPLACED_REFERENCE_NAME}), position is undefined.
+     * </p>
+     *
+     * @return 1-based start position on the reference; undefined if {@link #getReferenceName()}
+     * returns {@link #UNPLACED_REFERENCE_NAME}.
+     */
+    long getStart();
+
+    /**
+     * Gets the end position on the reference (1-based fully-closed) if present.
+     *
+     * @return 1-based fully-closed if present; empty optional otherwise.
+     *
+     * @implSpec if present, end position should be downstream te start position ({@code getStart()
+     * > getEnd()}).
+     * @see #getStart()
+     */
+    OptionalLong getEnd();

```

Comment:
Why make this optional?  In a 1-based system a point just has end == start.  Better to just require it I think and a point class can avoid storing it if they want.  Adding an optional adds boxing overhead and complexity.

Reply:
Returned as long now.